File: Easy/2144_Minimum_Cost_of_Buying_Candies_With_Discount/Solution.java
import java.util.Arrays;


public class Solution {
    public int minimumCost(int[] cost) {
        Arrays.sort(cost);

        int ans = 0;
        int cnt = 0;

        for (int i = cost.length - 1; i >= 0; i--) {
            cnt++;

            if (cnt % 3 != 0) {
                ans += cost[i];
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] cost1 = {1, 2, 3};
        System.out.println(solution.minimumCost(cost1)); // Output: 5

        // test cases 2
        int[] cost2 = {6, 5, 7, 9, 2, 2};
        System.out.println(solution.minimumCost(cost2)); // Output: 23

        // test cases 3
        int[] cost3 = {5, 5};
        System.out.println(solution.minimumCost(cost3)); // Output: 10
        
    }
}

File: Easy/2144_Minimum_Cost_of_Buying_Candies_With_Discount/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int minimumCost(vector<int>& cost) {
        int n = cost.size();
        int totalCost = 0;

        if (n < 3) {
            for (int c : cost) totalCost += c;
            return  totalCost;
        }

        sort(cost.begin(), cost.end(), greater<int>());

        for (int i = 0; i < n; i++) {
            if (i % 3 == 2) continue;
            totalCost += cost[i];
        }

        return totalCost;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> cost1 = {1, 2, 3};
    cout << solution.minimumCost(cost1) << endl; // Output: 5

    // test cases 2
    vector<int> cost2 = {6, 5, 7, 9, 2, 2};
    cout << solution.minimumCost(cost2) << endl; // Output: 23

    // test cases 3
    vector<int> cost3 = {5, 5};
    cout << solution.minimumCost(cost3) << endl; // Output: 10

    return 0;
}